logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueReader.java

```diff
@@ -344,8 +357,13 @@ private void removeSegmentsBefore(Path validSegment) throws IOException {
                 .thenComparingInt(DeadLetterQueueUtils::extractSegmentId);
 
         try (final Stream<Path> segmentFiles = listSegmentPaths(queuePath)) {
-            segmentFiles.filter(p -> fileTimeAndName.compare(p, validSegment) < 0)
-                  .forEach(this::deleteSegment);
+            LongSummaryStatistics deletionStats = segmentFiles.filter(p -> fileTimeAndName.compare(p, validSegment) < 0)
+                    .mapToLong(this::deleteSegment)
+                    .summaryStatistics();
+
+            // update consumed metrics
+            consumedSegments.add(deletionStats.getCount());
```

Comment:
If `deleteSegment` fails (ie, throws, catches and returns 0), this will still add to the count.